src/main/java/ca/uhn/fhir/jpa/starter/dotBase/services/Authorization.java

```diff
@@ -0,0 +1,92 @@
+import org.hl7.fhir.r4.model.Procedure;
+import org.hl7.fhir.r4.model.Procedure.ProcedureStatus;
+import org.hl7.fhir.r4.model.ResourceType;
+
+public class Authorization implements IConsentService {
+  private static final org.slf4j.Logger ourLog = org.slf4j.LoggerFactory.getLogger(Authorization.class);
+
+  private static final Set<String> DRAFT_RESOURCE_TYPES;
+
+  //TODO: delimit resourceTypes or include all types? If so missing draft resourceTypes?
+  static {
+    DRAFT_RESOURCE_TYPES = new HashSet<String>();
+    DRAFT_RESOURCE_TYPES.add(ResourceType.Procedure.toString());
+    DRAFT_RESOURCE_TYPES.add(ResourceType.QuestionnaireResponse.toString());
+    DRAFT_RESOURCE_TYPES.add(ResourceType.Condition.toString());
+  }
+
+  /**
+   * Invoked once at the start of every request
+   */
+  @Override
+  public ConsentOutcome startOperation(RequestDetails theRequestDetails, IConsentContextServices theContextServices) {
+    return ConsentOutcome.PROCEED;
+  }
+
+  /**
+   * Can a given resource be returned to the user?
+   */
+  @Override
+  public ConsentOutcome canSeeResource(RequestDetails theRequestDetails, IBaseResource theResource,
+    IConsentContextServices theContextServices
+  ) {
+    if (theRequestDetails.getRequestType() == RequestTypeEnum.GET && isDraftResource(theResource)) {
+      return isAuthorizedRequester(theRequestDetails, (DomainResource) theResource)
+        ? ConsentOutcome.AUTHORIZED
+        : ConsentOutcome.REJECT;
+    }
+    return ConsentOutcome.AUTHORIZED;
+  }
+
+  /**
+   * Modify resources that are being shown to the user
+   */
+  @Override
+  public ConsentOutcome willSeeResource(RequestDetails theRequestDetails, IBaseResource theResource,
+      IConsentContextServices theContextServices) {
+    return ConsentOutcome.AUTHORIZED;
+  }
+
+  private boolean isDraftResource(IBaseResource theResource) {
+    if (DRAFT_RESOURCE_TYPES.contains(theResource.fhirType())) {
+      return isDraft(theResource);
+    }
+    return false;
+  }
+
+  private static boolean isDraft(IBaseResource theResource) {
+    if (theResource instanceof Procedure) {
+      Procedure procedure = (Procedure) theResource;
+      return procedure.getStatus() == ProcedureStatus.INPROGRESS;
+    }
+    return ((Element) theResource).getExtensionByUrl("https://simplifier.net/dot.base/draft-action")!=null;
+  }
+
+  private boolean isAuthorizedRequester(RequestDetails theRequestDetails, DomainResource theResource) {
+    String requestingUser = (String) theRequestDetails.getAttribute("_username");
+    return isResourceEditor(theResource, requestingUser);
+  }
+
+  private static boolean isResourceEditor(DomainResource theResource,String requestingUser) {
+    Extension usernameExtension = theResource.getExtensionByUrl("https://simplifier.net/dot.base/resource-editor-username");
+    if (usernameExtension != null) {
+      return usernameExtension.getValue().toString().equals(requestingUser) 
+        ? true
+        : false;
```

Comment:
The `equals` function should return a boolean in any case. There is no need for a ternary operator.
```suggestion
      return usernameExtension.getValue().toString().equals(requestingUser);
```